trin-history/src/jsonrpc.rs

```diff
@@ -243,33 +247,14 @@ impl HistoryRequestHandler {
         if !is_trace {
             return Ok(Value::String(hex::encode(content)));
         }
-
-        // Construct trace response
-        let closest_nodes: Vec<NodeInfo> = closest_nodes
-            .iter()
-            // Skip over enrs that are unable to be looked up
-            .filter_map(|node_id| {
-                let content_id = content_key.content_id();
-                // Return None for enrs that cannot be looked up
-                match self.network.overlay.discovery.find_enr(node_id) {
-                    Some(enr) => XorMetric::distance(&content_id, &enr.node_id().raw())
-                        .log2()
-                        .map(|distance| {
-                            let distance = Distance::from(distance);
-                            NodeInfo {
-                                enr: EthEnr::from_str(&enr.to_base64()).unwrap(),
-                                distance,
-                            }
-                        }),
-                    None => None,
-                }
-            })
-            .collect();
-
-        Ok(json!(TraceContentInfo {
-            content: content.into(),
-            route: closest_nodes,
-        }))
+        if let Some(trace) = trace {
+            Ok(json!({
+                "content": hex_encode(content),
+                "trace": trace,
+            }))
+        } else {
+            Err("Content query trace requested but none provided.".to_owned())
+        }
```

Comment:
if we cannot find a clean way to separate the trace and non-trace queries, then I think it would be preferable to always build the trace throughout the query so that we don't have to deal with this `None` possibility.

Reply:
I'm hesitant to switch it back because of the tradeoff that previously if the query returned an error it also constructed and returned an empty query trace, which is kind of obnoxious. I think that between the two this is cleaner for that reason. If we can find a third even cleaner option in the future let's do that.